streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamSlidingWindowAggregate.java

```diff
@@ -467,27 +464,26 @@ private boolean rightWindowIsNotEmpty(final ValueAndTimestamp<VAgg> rightWinAgg,
 
         private void updateWindowAndForward(final Window window,
                                             final ValueAndTimestamp<VAgg> valueAndTime,
-                                            final KIn key,
-                                            final VIn value,
+                                            final Record<KIn, VIn> record,
                                             final long closeTime,
                                             final long inputRecordTimestamp) {
             final long windowStart = window.start();
             final long windowEnd = window.end();
             if (windowEnd > closeTime) {
                 //get aggregate from existing window
                 final VAgg oldAgg = getValueOrNull(valueAndTime);
-                final VAgg newAgg = aggregator.apply(key, value, oldAgg);
+                final VAgg newAgg = aggregator.apply(record.key(), record.value(), oldAgg);
 
                 final long newTimestamp = oldAgg == null ? inputRecordTimestamp : Math.max(inputRecordTimestamp, valueAndTime.timestamp());
                 windowStore.put(
-                    key,
+                    record.key(),
                     ValueAndTimestamp.make(newAgg, newTimestamp),
                     windowStart);
                 tupleForwarder.maybeForward(
-                    new Windowed<>(key, window),
-                    newAgg,
-                    sendOldValues ? oldAgg : null,
-                    newTimestamp);
+                    new Record<>(
+                        new Windowed<>(record.key(), window),
+                        new Change<>(newAgg, sendOldValues ? oldAgg : null),
+                        newTimestamp));
```

Comment:
IIRC, we were thinking about pushing the Record down here specifically so we could keep its headers. It's also fine if you want to not worry about it right now, though.

Reply:
for sure! fixing it.